lib/translator/common.h

```diff
@@ -123,5 +123,22 @@ bool WriteValue(const T& data, Span<uint8_t> out) {
   return true;
 }
 
+// Returns a pointer of type T pointing to buf.data()
+// Uses placement new operator to garentee memory safety
+// Validates alignment and size of buffer
```

Comment:
This should state that it creates a new `T`; trying to access an existing one would be undefined behavior. Typo in "guarantee", also.